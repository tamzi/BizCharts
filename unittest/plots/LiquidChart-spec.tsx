import React from 'react';
import LiquidChart from '../../src/plots/LiquidChart';
import { render, cleanup } from '@testing-library/react';

describe('Plots-LiquidChart', () => {
  test('兼容旧版LiquidChart', () => {
    let chart = null;
    render(<LiquidChart 
      title={{
        visible: true,
        text: '水波图',
      }}
      max={10000}
      value={5639}
      // percent={0.5}
      onGetG2Instance={
        (c) => chart = c
      }
    />);
    expect(chart.options.percent).toBe(5639 / 10000);
    expect(chart.options).toMatchSnapshot();
    cleanup()
  })
  test('新版使用', () => {
    let chart = null;
    render(<LiquidChart 
      title={{
        visible: true,
        text: '水波图',
      }}
      percent={0.5}
      outline={{
        border: 4,
        distance: 8,
      }}
      wave={{
        length: 128,
      }}
      onGetG2Instance={
        (c) => chart = c
      }
    />);
    expect(chart.options).toMatchSnapshot();
    cleanup()
  })
})
