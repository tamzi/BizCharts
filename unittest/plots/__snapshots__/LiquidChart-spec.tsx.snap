// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Plots-LiquidChart 兼容旧版LiquidChart 1`] = `
Object {
  "appendPadding": Array [
    10,
    5,
    10,
    10,
  ],
  "autoFit": true,
  "color": "#6a99f9",
  "data": 0.5639,
  "outline": Object {
    "border": 2,
    "distance": 0,
  },
  "percent": 0.5639,
  "radius": 0.9,
  "statistic": Object {
    "content": Object {
      "formatter": [Function],
      "style": Object {
        "fontSize": "30px",
        "lineHeight": "30px",
        "opacity": 0.75,
        "textAlign": "center",
      },
    },
    "title": false,
  },
  "wave": Object {
    "count": 3,
    "length": 192,
  },
}
`;

exports[`Plots-LiquidChart 新版使用 1`] = `
Object {
  "appendPadding": Array [
    10,
    5,
    10,
    10,
  ],
  "autoFit": true,
  "color": "#6a99f9",
  "data": 0.5,
  "outline": Object {
    "border": 4,
    "distance": 8,
  },
  "percent": 0.5,
  "radius": 0.9,
  "statistic": Object {
    "content": Object {
      "formatter": [Function],
      "style": Object {
        "fontSize": "30px",
        "lineHeight": "30px",
        "opacity": 0.75,
        "textAlign": "center",
      },
    },
    "title": false,
  },
  "wave": Object {
    "count": 3,
    "length": 128,
  },
}
`;
